segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/BaseChunkStorageProvider.java

```diff
@@ -0,0 +1,592 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.segmentstore.storage.chunklayer;
+
+import com.google.common.base.Preconditions;
+import io.pravega.common.LoggerHelpers;
+import io.pravega.common.Timer;
+import io.pravega.shared.MetricsNames;
+import io.pravega.shared.metrics.Counter;
+import io.pravega.shared.metrics.MetricsProvider;
+import io.pravega.shared.metrics.OpStatsLogger;
+import io.pravega.shared.metrics.StatsLogger;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.time.Duration;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * Base implementation of {@link ChunkStorageProvider}.
+ * It implements common functionality for
+ * Delegates to specific implemntations by calling various abstract methods which must be overridden in derived classes.
+ */
+public abstract class BaseChunkStorageProvider implements ChunkStorageProvider {
+
+    protected static final org.slf4j.Logger log = org.slf4j.LoggerFactory.getLogger(BaseChunkStorageProvider.class);
+
+    protected static final StatsLogger STATS_LOGGER = MetricsProvider.createStatsLogger("BaseChunkStorageProvider");
+
+
+    protected static final OpStatsLogger READ_LATENCY = STATS_LOGGER.createStats(MetricsNames.STORAGE_READ_LATENCY);
```

Comment:
Put these metrics in a separate, static class. Please follow the pattern used throughout other parts of the system.

Reply:
done.